numberformat_proposed.html

```diff
@@ -463,6 +482,84 @@ <h1>UnwrapNumberFormat( _nf_ )</h1>
+        1. Let _currency_ be ? GetOption(_options_, `"currency"`, `"string"`, *undefined*, *undefined*).
+        1. Let _currencyDisplay_ be ? GetOption(_options_, `"currencyDisplay"`, `"string"`, &laquo; `"code"`, `"symbol"`, `"narrowSymbol"`, `"name"` &raquo;, `"symbol"`).
+        1. Let _unit_ be ? GetOption(_options_, `"unit"`, `"string"`, *undefined*, *undefined*).
+        1. Let _unitDisplay_ be ? GetOption(_options_, `"currencyDisplay"`, `"string"`, &laquo; `"short"`, `"narrow"`, `"long"` &raquo;, `"short"`).
+        1. Let _currencySignDisplay_ be ? GetOption(_options_, `"currencySignDisplay"`, `"string"`, &laquo; `"standard"`, `"accounting"` &raquo;, `"standard"`).
+        1. If _style_ is `"decimal"`, then
+          1. Set _intlObj_.[[UnitType]] to `"none"`.
+        1. If _style_ is `"currency"`, then
+          1. If _currency_ is *undefined*, throw a *TypeError* exception.
+          1. If the result of IsWellFormedCurrencyCode(_currency_) is *false*, throw a *RangeError* exception.
+          1. Let _currency_ be the result of converting _currency_ to upper case as specified in <emu-xref href="#sec-case-sensitivity-and-case-mapping"></emu-xref>.
+          1. Set _intlObj_.[[UnitType]] to `"currency"`.
+          1. Set _intlObj_.[[Currency]] to _currency_.
+          1. Set _intlObj_.[[CurrencyDisplay]] to _currencyDisplay_.
+          1. Set _intlObj_.[[CurrencySignDisplay]] to _currencySignDisplay_.
+        1. If _style_ is `"percent"`, then
+          1. Let _unit_ be `"concentr-percent"`.
+        1. If _style_ is `"unit"` or _style_ is `"percent"`, then
+          1. If _unit_ is *undefined*, throw a *TypeError* exception.
+          1. If the result of IsWellFormedUnitIdentifier(_unit_) is *false*, throw a *RangeError* exception.
+          1. Set _intlObj_.[[UnitType]] to `"measure-unit"`.
+          1. Set _intlObj_.[[MeasureUnit]] to _unit_.
+          1. Set _intlObj_.[[MeasureUnitDisplay]] to _unitDisplay_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-getnumberformatpattern" aoid="GetNumberFormatPattern">
+      <h1>GetNumberFormatPattern ( _numberFormat_, _x_ )</h1>
+      <p>
+        The abstract operation GetNumberFormatPattern considers the resolved unit-related options in the number format object along with the final scaled and rounded number being formatted and returns a pattern, a String value as described in <emu-xref href="#sec-intl.numberformat-internal-slots"></emu-xref>. 
+      </p>
+      <emu-alg>
+        1. Let _localeData_ be %NumberFormat%.[[LocaleData]].
+        1. Let _dataLocale_ be _numberFormat_.[[DataLocale]].
+        1. Let _dataLocaleData_ be _localeData_.[[&lt;_dataLocale_&gt;]].
+        1. Let _patterns_ be _dataLocaleData_.[[patterns]].
+        1. Assert: _patterns_ is a record (see <emu-xref href="#sec-intl.numberformat-internal-slots"></emu-xref>).
+        1. TODO: Is there a spec defining the format of _patterns_ that needs to be updated, too?
```

Comment:
Do you mean the format of these nested records that include the patterns? I think this is what you'd want to update: https://tc39.github.io/ecma402/#sec-intl.numberformat-internal-slots